# Data Exploration

***

## Selecting Columns

In [2]:
# Loading Python Libraries

import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

#### Loading Dataset

In [3]:
# Films Dataset

films = pd.read_csv("D:/documents/xlsx/films.csv")

#### Showing first few rows of the dataset

In [3]:
films.head()

id                                             title  release_year  \
0   1  Intolerance: Love's Struggle Throughout the Ages        1916.0   
1   2                    Over the Hill to the Poorhouse        1920.0   
2   3                                    The Big Parade        1925.0   
3   4                                        Metropolis        1927.0   
4   5                                     Pandora's Box        1929.0   

   country  duration language certification      gross     budget  
0      USA     123.0      NaN     Not Rated        NaN   385907.0  
1      USA     110.0      NaN           NaN  3000000.0   100000.0  
2      USA     151.0      NaN     Not Rated        NaN   245000.0  
3  Germany     145.0   German     Not Rated    26435.0  6000000.0  
4  Germany     110.0   German     Not Rated     9950.0        NaN

#### Checking the variables

In [5]:
films.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4968 entries, 0 to 4967
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             4968 non-null   int64  
 1   title          4968 non-null   object 
 2   release_year   4926 non-null   float64
 3   country        4966 non-null   object 
 4   duration       4955 non-null   float64
 5   language       4957 non-null   object 
 6   certification  4666 non-null   object 
 7   gross          4158 non-null   float64
 8   budget         4538 non-null   float64
dtypes: float64(4), int64(1), object(4)
memory usage: 349.4+ KB


#### Checking Missing Values

In [6]:
films.isna().any()

id               False
title            False
release_year      True
country           True
duration          True
language          True
certification     True
gross             True
budget            True
dtype: bool

**Missing Value Variables**:- release_year, country, duration, language, certification, gross, budget

In [7]:
# Select title column from table films

films[["title"]]

title
0     Intolerance: Love's Struggle Throughout the Ages
1                       Over the Hill to the Poorhouse
2                                       The Big Parade
3                                           Metropolis
4                                        Pandora's Box
...                                                ...
4963                                       Unforgotten
4964                                             Wings
4965                                        Wolf Creek
4966                                 Wuthering Heights
4967                           Yu-Gi-Oh! Duel Monsters

[4968 rows x 1 columns]

In [9]:
# Select release year from table
films.loc[:, ["release_year"]]

release_year
0           1916.0
1           1920.0
2           1925.0
3           1927.0
4           1929.0
...            ...
4963           NaN
4964           NaN
4965           NaN
4966           NaN
4967           NaN

[4968 rows x 1 columns]

In [10]:
# Select first 10 columns
films.head(10)

id                                             title  release_year  \
0   1  Intolerance: Love's Struggle Throughout the Ages        1916.0   
1   2                    Over the Hill to the Poorhouse        1920.0   
2   3                                    The Big Parade        1925.0   
3   4                                        Metropolis        1927.0   
4   5                                     Pandora's Box        1929.0   
5   6                               The Broadway Melody        1929.0   
6   7                                     Hell's Angels        1930.0   
7   8                                A Farewell to Arms        1932.0   
8   9                                       42nd Street        1933.0   
9  10                                She Done Him Wrong        1933.0   

   country  duration language certification      gross     budget  
0      USA     123.0      NaN     Not Rated        NaN   385907.0  
1      USA     110.0      NaN           NaN  3000000.0   100000.0  
2      USA     151.0      NaN     Not Rated        NaN   245000.0  
3  Germany     145.0   German     Not Rated    26435.0  6000000.0  
4  Germany     110.0   German     Not Rated     9950.0        NaN  
5      USA     100.0  English        Passed  2808000.0   379000.0  
6      USA      96.0  English        Passed        NaN  3950000.0  
7      USA      79.0  English       Unrated        NaN   800000.0  
8      USA      89.0  English       Unrated  2300000.0   439000.0  
9      USA      66.0  English      Approved        NaN   200000.0

In [27]:
# How many records are contained in films table
films.shape

(4968, 9)

total 4968 rows and 9 columns are contained.

---

## Filtering Rows

In [33]:
# Get all the details for all the films released in 2016
released_movies_2016 = films[films["release_year"] == 2016]
released_movies_2016.head(7)

id                            title  release_year country  duration  \
4820  4821              10 Cloverfield Lane        2016.0     USA     104.0   
4821  4822                         13 Hours        2016.0     USA     144.0   
4822  4823      A Beginner's Guide to Snuff        2016.0     USA      87.0   
4823  4824                          Airlift        2016.0   India     130.0   
4824  4825  Alice Through the Looking Glass        2016.0     USA     113.0   
4825  4826                        Allegiant        2016.0     USA     120.0   
4826  4827   Alleluia! The Devil's Carnival        2016.0     USA      97.0   

     language certification       gross       budget  
4820  English         PG-13  71897215.0   15000000.0  
4821  English             R  52822418.0   50000000.0  
4822  English           NaN         NaN          NaN  
4823    Hindi           NaN         NaN    4400000.0  
4824  English            PG  76846624.0  170000000.0  
4825  English         PG-13  66002193.0  110000000.0  
4826  English           NaN         NaN     500000.0

In [23]:
# Get the number of films released before 2016

filter_release_movie = films["release_year"] < 2016
released_before_2016 = films.loc[:, ["release_year"]] [filter_release_movie].value_counts(sort = False)
released_before_2016


release_year
1916.0            1
1920.0            1
1925.0            1
1927.0            1
1929.0            2
               ... 
2011.0          224
2012.0          220
2013.0          236
2014.0          252
2015.0          226
Length: 90, dtype: int64

In [13]:
# Get the title and release year of films released after 2000
fitler_movies_after_2000 = films["release_year"] > 2000
two_var = films.loc[:, ["title", "release_year"]]
released_after_2000 = two_var[fitler_movies_after_2000]
released_after_2000.head(7)

title  release_year
1508                    15 Minutes        2001.0
1509       3000 Miles to Graceland        2001.0
1510              A Beautiful Mind        2001.0
1511               A Knight's Tale        2001.0
1512  A.I. Artificial Intelligence        2001.0
1513                           Ali        2001.0
1514                   Alias Betty        2001.0

In [ ]:
## Get all the details for all French language films
